app/controllers/CommentsController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+//require_once 'models/CommentModel.php';
+//require_once 'models/PostModel.php';
+namespace app\controllers;
+
+use app\models\CommentModel;
+use app\models\PostModel;
+use app\utils\View;
+
+class CommentsController {
+    private CommentModel $commentModel;
+    private PostModel $postModel;
+
+    public function __construct() {
+        $this->commentModel = new CommentModel(); // Initialize the PostModel instance
+        $this->postModel = new PostModel();
+    }
+
+    public function store($postId) {
+        $name = trim($_POST['name'] ?? '');
+        // Check if the name is empty
+        if (empty($name)) {
+            $name = 'Anonymous'; // Set default name
+        }
+        $message = $_POST['message'];
+        $userId = $_SESSION['user_id'];
+        $role = $_SESSION['role'];
+//        var_dump($name);
+        if (strlen($message) > 50) {
+            $_SESSION['error'] = "Comment exceeds the maximum character limit.";
+            if ($role === 'admin') {
+                header('Location: /admin/posts/' . $postId);
+                exit;
+            }
+            header('Location: /posts/' . $postId);
+            exit;
+        }
+
+        $this->commentModel->addComment($postId, $name, $message, $userId);
```

Comment:
@tony-tang-aligent I'm going to use this line as an alternative suggestion to class naming.  Your current code:

```
$this->commentModel->addComment($postId, $name, $message, $userId);
```

I for readability I would suggest that we remove the phrase `Model` from the model class names (ie. call it just `Comment` instead of `CommentModel`).

Secondly I'd suggest renaming the methods inside the `Comment` model to remove the `Comment` part.  ie. just call the method `add` instead of `addComment`

With those changes in mind the code could be re-written

```
$this->Comment->add($postId, $name, $message, $userId);

I'm not expecting you to go back and re-factor all of you classes (though you're welcome to!), this is more a general suggestion (I'd note that Magento isn't overly consistent in this approach either!)
```

Reply:
I'll add a little further to this - could you have an `Entity` class which contained `create`, `update`, `delete`, etc. functions which could be shared between your models?